widget/toolbar.go

```diff
@@ -37,7 +38,17 @@ func (t *ToolbarAction) SetIcon(icon fyne.Resource) {
 
 // NewToolbarAction returns a new push button style ToolbarItem
 func NewToolbarAction(icon fyne.Resource, onActivated func()) *ToolbarAction {
-	return &ToolbarAction{icon, onActivated}
+	toolbarAction := &ToolbarAction{
+		Icon:        icon,
+		OnActivated: onActivated,
+	}
+	return toolbarAction
+}
+
+// SetDisabled updated the disabled property of ToolbarAction
+func (t *ToolbarAction) SetDisabled(disabled bool) {
+	t.disabled = disabled
+	t.ToolbarObject().Refresh()
```

Comment:
This doesn't seem right - it creates a new object visual and then refreshes it instead of refreshing the current UI.

Reply:
Thanks for review. Does t.Refresh() makes sense? That will re-render same element.